proofs/flatten.go

```diff
@@ -185,29 +186,31 @@ func (f *messageFlattener) appendLeaf(prop Property, value string, salt []byte,
 	f.leaves = append(f.leaves, leaf)
 }
 
-func (f *messageFlattener) valueToString(value interface{}) (s string, err error) {
+func (f *messageFlattener) valueToBytesArray(value interface{}) (b []byte, err error) {
 	switch v := value.(type) {
 	case nil:
-		return "", nil
+		return []byte{}, nil
 	case string:
-		return v, nil
+		return []byte(v), nil
 	case int8, int16, int32, int64, uint8, uint16, uint32, uint64:
-		return fmt.Sprint(value), nil
+		return []byte(fmt.Sprint(v)), nil
```

Comment:
This actually encodes the integer to string and then to bytes. Instead an integer should be converted straight to binary. See: https://stackoverflow.com/questions/16888357/convert-an-integer-to-a-byte-array